### PR TITLE
Clarify license and cleanup

### PR DESCRIPTION
**Pull request by Panquesito7**
Things added/changed:

- Clarify license files.
  - `LICENSE.txt` contains the used licenses in both source code and media.
  - `COPYING.LESSER` is the full [GNU LGPLv2.1](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html) license text.
  - `AUTHORS.md` contains the source code authors and specific media files as well as their licenses.
- Minor cleanup in various files.

## To do

Ready for review.

## How to test

Not applicable.

### CODE REVIEW

**Comment by SmallJoker** [comment]
Why does the file contain an entire copy of the license text? Wouldn't it suffice to keep the short "header" version?

**Comment by Panquesito7** [comment]
> Why does the file contain an entire copy of the license text? Wouldn't it suffice to keep the short "header" version?

Do you mean the LGPLv2.1+ license?

**Comment by SmallJoker** [comment]
Yes

**Comment by Panquesito7** [comment]
> Yes

I believe a copy of the license should be present in the repository. If you prefer, though, we can either add a `COPYING.LESSER` file containing the license and add the header information in the license file, or just keep the header information.

**Comment by SmallJoker** [comment]
> Must
>> Including the full text of license in modified software.

Source: https://www.tldrlegal.com/license/gnu-lesser-general-public-license-v2-1-lgpl-2-1

I see. The main concern I have is that it's not as easy any more to actually *find* the license of a certain part of the project when it's buried between full-fat license texts. Hence I would very much prefer to have the license text in a separate file.

**Comment by Panquesito7** [comment]
> > Must
> > > Including the full text of license in modified software.
> 
> Source: https://www.tldrlegal.com/license/gnu-lesser-general-public-license-v2-1-lgpl-2-1
> 
> I see. The main concern I have is that it's not as easy any more to actually _find_ the license of a certain part of the project when it's buried between full-fat license texts. Hence I would very much prefer to have the license text in a separate file.

Yep, that sounds like a better approach. I'll put it in a separate `COPYING.LESSER` file.

**Comment by Panquesito7** [comment]
Updated (see first message for more information).

File: init.lua [hunk 1]
```diff
@@ -1,6 +1,6 @@
 --[[
-Allows players to request from another player to be teleported to them, and do much more.
-Copyright (C) 2014-2023 ChaosWormz and contributors
+Allows players to request from another player to be teleported to them. Includes many more teleporting features. Built for Minetest.
```
Comment (SmallJoker) [inline init.lua:2]: Please break the line

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
 # Contributing
 
-Welcome to the [Teleport Request](https://github.com/minetest-mods/teleport-request) mod! Before contributing, **please make sure** you have read the whole guidelines mentioned below. If you have any doubts about these guidelines, please open [an issue](https://github.com/minetest-mods/teleport-request/issues/new), and clearly state your concerns.
+Welcome to the [Teleport Request](https://github.com/minetest-mods/teleport-request) mod! Before contributing, **please make sure** you have read the guidelines mentioned below. If you have any doubts about these guidelines, please open [an issue](https://github.com/minetest-mods/teleport-request/issues/new), and clearly state your concerns. Thank you for your interest in contributing!
```
Comment (SmallJoker) [inline CONTRIBUTING.md:3]: Please break the line while you're at it.

File: .luacheckrc [hunk 3]
```diff
@@ -1,6 +1,6 @@
 unused_args = false
 allow_defined_top = true
-max_line_length = 999
+max_line_length = false
```
Comment (SmallJoker) [inline .luacheckrc:3]: Why?

**Comment by SmallJoker** [comment]
Sorry, I didn't catch the translation issue last time, thus another review iteration. Aside from that the changes look good :+1:.

**Comment by Panquesito7** [comment]
Thanks for the review! Now, another change that I missed. According to [this article](https://www.gnu.org/licenses/gpl-howto.en.html#license-notices), we should change the copyright notices on each file to say `Teleport Request...` instead of `This library` because there are multiple source files. Should we change that as well?

**Comment by SmallJoker** [comment]
@Panquesito7 I don't think anyone cares about such formalities in one of the thousand Minetest mods.

**Pull request merged**